GDScript style guide: Improve type inference section (#6661)
godotengine/godot-docs: tutorials/scripting/gdscript/gdscript_styleguide.rst

Inferred types
~~~~~~~~~~~~~~

In most cases you can let the compiler infer the type, using ``:=``.
Prefer ``:=`` when the type is written on the same line as the assignment,
otherwise prefer writing the type explicitly.

**Good**:

.. rst-class:: code-example-good

::

>    var health: int = 0 # The type should be stated explicitly.
>    var direction := Vector3(1, 2, 3) # The type is obviously Vector3.

Reviewer: ```suggestion
   var health: int = 0 # The type can be int or float, and thus should be stated explicitly.
   var direction := Vector3(1, 2, 3) # The type is clearly inferred as Vector3.
```

> Don't exclude the type hint when it's not obvious, and
> don't include the type hint when it's redundant.

Reviewer: ```suggestion
Include the type hint when the type is ambiguous, and
omit the type hint when it's redundant.
```
